File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/tools/file_watchers/FileWatcherAdvancedOptionsSettings.java
package com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableFileWatcherAdvancedOptionsSettings.class)
public interface FileWatcherAdvancedOptionsSettings {

  Optional<Boolean> autoSaveEditedFilesToTriggerTheWatcher();

  Optional<Boolean> triggerTheWatcherRegardlessOfSyntaxErrors();

  Optional<Boolean> triggerTheWatcherOnExternalChanges();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/tools/file_watchers/FileWatcherAdvancedOptionsSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.tools.file_watchers

import com.fasterxml.jackson.annotation.JsonProperty

data class FileWatcherAdvancedOptionsSettings(

    @JsonProperty("Auto-save edited files to trigger the watcher")
    val autoSaveEditedFilesToTriggerTheWatcher: Boolean? = null,

    @JsonProperty("Trigger the watcher regardless of syntax errors")
    val triggerTheWatcherRegardlessOfSyntaxErrors: Boolean? = null,

    @JsonProperty("Trigger the watcher on external changes")
    val triggerTheWatcherOnExternalChanges: Boolean? = null
)
